src/Adapter/Shop/Context.php

```diff
@@ -72,6 +72,23 @@ public function getContextListShopID($share = false)
         return Shop::getContextListShopID($share);
     }
 
+    /**
+     * Return the result of getContextListShopID() for customers usecase
+     * This handles the "multishop sharing customer" feature setting
+     *
+     * @return array
+     */
+    public function getContextListShopIDUsingCustomerSharingSettings()
+    {
+        $groupSettings = Shop::getGroupFromShop(Shop::getContextShopID(), false);
+
+        if ($groupSettings['share_customer']) {
+            return Shop::getContextListShopID(Shop::SHARE_CUSTOMER);
```

Comment:
Maybe we should manage the other two sharing types since we're at it ^^
Shop::SHARE_ORDER and SHOP::SHARE_STOCK
cd https://github.com/PrestaShop/PrestaShop/blob/2776a60d01dc3dac4fbfeb02b8bb67daedbed574/classes/shop/Shop.php#L900

Reply:
I wonder if we should also take a look at other places where this method is used like:
https://github.com/PrestaShop/PrestaShop/blob/2776a60d01dc3dac4fbfeb02b8bb67daedbed574/src/Adapter/Module/ModuleDataProvider.php#L167

But I guess this bug only occurs for entities that have sharing capabilities, so Customer, Order, Stock Then we must be careful about (develop branch only):

https://github.com/PrestaShop/PrestaShop/blob/2776a60d01dc3dac4fbfeb02b8bb67daedbed574/src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml#L202